# SI 618 WN 2024 — Project Part II: Analysis

**Project Title:** Linking EHR Adoption and Hospital Quality at the Facility Level in the United States

**Team Members:** Haichao Min(haichaom), Yuanyu Lin (yuanlin)

### Introduction

This notebook constitutes Part 2 of our project, building directly on the merged and cleaned dataset we finalized in Part 1. In the first phase, we successfully combined two key CMS datasets: the **Hospital General Information** (providing hospital characteristics and quality ratings) and the **Promoting Interoperability** (providing facility-level EHR technology data).

The goal of this section is to conduct a formal statistical analysis to extract insights and find significant associations within our data. As per the course guidelines, our focus is on **statistical analysis (Part 2)**, not predictive modeling (which is reserved for Part 3). Therefore, our objective is to identify and quantify relationships, not to determine causation or predictive power.

Our analysis will be guided by the three "real-world" questions we posed in Part 1:

1. **Do hospitals with a certified EHR system report higher overall quality scores compared to those without a reported system?**

2. **Is there a discernible difference in quality outcomes among hospitals using different major EHR vendors (e.g., Epic vs. Cerner vs. MEDITECH)?**

3. **Are specific types of hospitals, such as smaller or rural "Critical Access Hospitals," lagging in EHR adoption, and does this correlate with their reported outcomes?**

To investigate these questions, this notebook will proceed through each required component of the analysis: **Descriptive Statistics**, **Inferential Statistics** (t-tests, ANOVA, Chi-Squared), **Graphical Analysis**, **Comparative Analysis**, and **Multivariate Analysis**. Finally, we will **Synthesize** our findings to provide statistical answers to our research questions and **Reflect** on the discoveries and limitations of this analysis.


### Descriptive Statistics

In [2]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

In [3]:
# Load the dataset
df = pd.read_csv('Project_datasets/clean_df.csv')
df.head()

Facility ID                  Facility Name_x                   Address_x  \
0      010001  SOUTHEAST HEALTH MEDICAL CENTER      1108 ROSS CLARK CIRCLE   
1      010005         MARSHALL MEDICAL CENTERS  2505 U S HIGHWAY 431 NORTH   
2      010006     NORTH ALABAMA MEDICAL CENTER         1701 VETERANS DRIVE   
3      010007         MIZELL MEMORIAL HOSPITAL               702 N MAIN ST   
4      010008      CRENSHAW COMMUNITY HOSPITAL         101 HOSPITAL CIRCLE   

  City/Town_x State_x  ZIP Code_x County/Parish_x Telephone Number_x  \
0      DOTHAN      AL       36301         HOUSTON     (334) 793-8701   
1        BOAZ      AL       35957        MARSHALL     (256) 593-8310   
2    FLORENCE      AL       35630      LAUDERDALE     (256) 768-8400   
3         OPP      AL       36467       COVINGTON     (334) 493-3541   
4     LUVERNE      AL       36049        CRENSHAW     (334) 335-3374   

          Hospital Type                           Hospital Ownership  ...  \
0  Acute Care Hospitals  Government - Hospital District or Authority  ...   
1  Acute Care Hospitals  Government - Hospital District or Authority  ...   
2  Acute Care Hospitals                                  Proprietary  ...   
3  Acute Care Hospitals               Voluntary non-profit - Private  ...   
4  Acute Care Hospitals                                  Proprietary  ...   

  ZIP Code_y County/Parish_y Telephone Number_y  CEHRT ID  \
0        NaN             NaN                NaN       NaN   
1        NaN             NaN                NaN       NaN   
2        NaN             NaN                NaN       NaN   
3        NaN             NaN                NaN       NaN   
4        NaN             NaN                NaN       NaN   

  Meets criteria for promoting interoperability of EHRs Start Date End Date  \
0                                                NaN           NaN      NaN   
1                                                NaN           NaN      NaN   
2                                                NaN           NaN      NaN   
3                                                NaN           NaN      NaN   
4                                                NaN           NaN      NaN   

  rating_numeric EHR_Vendor  Has_EHR  
0            4.0      Other       No  
1            3.0      Other       No  
2            2.0      Other       No  
3            1.0      Other       No  
4            NaN      Other       No  

[5 rows x 52 columns]

In [4]:
# 1a. Analysis the main outcome variable: Hospital Overall Rating (rating_numeric)
print("--- Descriptive Statistics for Hospital Overall Rating ---")
print(df['rating_numeric'].describe())

# 1b. Analysis the main categorial variables
print("\n--- Descriptive Statistics: EHR Adoption Status ('Has_EHR') (in %) ---")
print(df['Has_EHR'].value_counts(normalize=True) * 100)

print("\n--- Descriptive Statistics: EHR Vendor Distribution ('EHR_Vendor') (in %) ---")
print(df['EHR_Vendor'].value_counts(normalize=True) * 100)

print("\n--- Descriptive Statistics: Top 5 Hospital Types ('Hospital Type') ---")
print(df['Hospital Type'].value_counts().head(5))

--- Descriptive Statistics for Hospital Overall Rating ---
count    2872.000000
mean        3.082521
std         1.099926
min         1.000000
25%         2.000000
50%         3.000000
75%         4.000000
max         5.000000
Name: rating_numeric, dtype: float64

--- Descriptive Statistics: EHR Adoption Status ('Has_EHR') (in %) ---
Has_EHR
Yes    63.445456
No     36.554544
Name: proportion, dtype: float64

--- Descriptive Statistics: EHR Vendor Distribution ('EHR_Vendor') (in %) ---
EHR_Vendor
Other                       87.716038
Epic Systems Corporation     6.560119
TruBridge, Inc.              3.493774
VigiLanz Corporation         2.230069
Name: proportion, dtype: float64

--- Descriptive Statistics: Top 5 Hospital Types ('Hospital Type') ---
Hospital Type
Acute Care Hospitals                    3127
Critical Access Hospitals               1370
Psychiatric                              626
Acute Care - Veterans Administration     132
Childrens                                 94
Nam

#### Observations

***Analysis: Hospital Overall Rating***

Based on our `describe()` output, we have several key findings for our primary outcome variable, `rating_numeric`:

* **Significant Missing Data:** The `count` is **2,872**. This is a critical finding. Out of our total ~5,381 hospitals, only 2,872 have a usable numeric rating. This indicates that our "overall rating" analysis will only apply to this subset, a key limitation we must discuss in our `Reflection`.
* **Central Tendency:** The `mean` (average) rating is **3.08**, and the `50%` (median) is **3.0**. These values are very close, suggesting a relatively symmetric distribution (not heavily skewed) centered around an "average" 3-star rating.
* **Spread:** The `std` (standard deviation) is 1.1, and the interquartile range (IQR) is from 2.0 (`25%`) to 4.0 (`75%`), which confirms a typical spread across the 1-5 star scale.

***Analysis: Categorical Variables***

Our main categorical variables provide the context for our analysis:

* **EHR Adoption Status (`Has_EHR`):** The split is **63.4% 'Yes'** and **36.6% 'No'**. This is an excellent, well-balanced split. It provides a substantial "No" group for comparison, making our first research question (comparing "Yes" vs. "No") statistically robust.
* **EHR Vendor Distribution (`EHR_Vendor`):** The most striking finding is that the **'Other' category accounts for 87.7%** of all hospitals. This "Other" group includes both the "No" hospitals (36.6%) and all unmapped smaller vendors (51.1%). This indicates the EHR market is highly fragmented, and our vendor-specific analysis will be a comparison between a few identified vendors (like `Epic Systems`) and this large, diverse "Other" group.
* **Hospital Types:** The output snippet (e.g., `Psychiatric`, `Childrens`) shows the diversity of hospital types in our dataset. As we will see in section `1e`, these specialized hospitals have unique characteristics (like 0% EHR adoption in this specific program, and `NaN` ratings) that are critical to our third research question.

In [5]:
# 1c. Descriptive statistics for Research Question 1
print("--- Descriptive Statistics: Hospital Ratings Grouped by 'Has_EHR' ---")
print(df.groupby('Has_EHR')['rating_numeric'].describe())

# 1d. Descriptive statistics for Research Question 2
print("\n--- Descriptive Statistics: Hospital Ratings Grouped by 'EHR_Vendor' ---")
print(df.groupby('EHR_Vendor')['rating_numeric'].describe())

# 1e. Descriptive statistics for Research Question 3
print("\n--- Stats for Q3 (Part 1): EHR Adoption Rate by Hospital Type ---")

# We use pd.crosstab to examine the relationship between 'Hospital Type' and 'EHR Adoption Status'
crosstab_q3 = pd.crosstab(df['Hospital Type'], df['Has_EHR'])
# Calculate adoption rates
crosstab_q3['Total'] = crosstab_q3['Yes'] + crosstab_q3['No']
crosstab_q3['Adoption_Rate (%)'] = (crosstab_q3['Yes'] / crosstab_q3['Total']) * 100

# Use ascending order to see which types are lagging behind
print(crosstab_q3.sort_values(by='Adoption_Rate (%)', ascending=True))

# Q3, Part 2: Does this affect their outcomes?
print("\n--- Stats for Q3 (Part 2): Hospital Ratings Grouped by 'Hospital Type' ---")

# Focus on the top 5 most common hospital types for clearer comparison
top_5_types = df['Hospital Type'].value_counts().head(5).index
df_top_5_types = df[df['Hospital Type'].isin(top_5_types)]

# Compare the rating distributions for these 5 types of hospitals
print(df_top_5_types.groupby('Hospital Type')['rating_numeric'].describe())

--- Descriptive Statistics: Hospital Ratings Grouped by 'Has_EHR' ---
          count      mean       std  min  25%  50%  75%  max
Has_EHR                                                     
No        682.0  3.137830  1.162965  1.0  2.0  3.0  4.0  5.0
Yes      2190.0  3.065297  1.079241  1.0  2.0  3.0  4.0  5.0

--- Descriptive Statistics: Hospital Ratings Grouped by 'EHR_Vendor' ---
                           count      mean       std  min  25%  50%  75%  max
EHR_Vendor                                                                   
Epic Systems Corporation   247.0  3.271255  1.025738  1.0  3.0  3.0  4.0  5.0
Other                     2474.0  3.081649  1.107177  1.0  2.0  3.0  4.0  5.0
TruBridge, Inc.             36.0  2.916667  1.130739  1.0  2.0  3.0  4.0  5.0
VigiLanz Corporation       115.0  2.747826  1.007372  1.0  2.0  3.0  3.0  5.0

--- Stats for Q3 (Part 1): EHR Adoption Rate by Hospital Type ---
Has_EHR                                No   Yes  Total  Adoption_Rate (%)
Hos

#### Observations

***Analysis: Research Question 1 (Has_EHR vs. Rating)***

Our analysis of section `1c` provides a direct answer to our first research question: "Do hospitals with a certified EHR system report higher overall quality scores?"

* **Key Finding:** The descriptive statistics show **no meaningful difference** in quality ratings between hospitals with and without a reported EHR system.
* **The Numbers:**
    * The `mean` (average) rating for the **'No'** group is **3.14**, which is surprisingly slightly *higher* than the `mean` for the **'Yes'** group (**3.07**).
    * The `50%` (median) for both groups is identical at **3.0**.
* **Conclusion:** This strongly supports our finding from Part 1's boxplot. At the descriptive level, our data does not support the hypothesis that simply having a certified EHR is associated with a higher hospital rating. This sets the stage for our `t-test` in the next section to confirm if this tiny difference is statistically significant (it likely isn't).

***Analysis: Research Question 2 (EHR\_Vendor vs. Rating)***

Our analysis of section `1d` explores our second question: "Is there a difference in outcomes among hospitals using different EHR vendors?"

* **Key Finding:** Yes, unlike the simple "Yes/No" comparison, the choice of EHR vendor appears to be associated with **clear differences** in average ratings.
* **The Numbers:**
    * `Epic Systems Corporation` hospitals have the highest average rating at **3.27**.
    * `VigiLanz Corporation` hospitals have the lowest average rating at **2.75**.
    * The large `Other` group (which includes the 'No' group) sits in the middle with a mean of **3.08**.
* **Conclusion:** This is a much more nuanced and interesting finding. It suggests that while *adoption* itself isn't a clear predictor of quality, the *specific system* a hospital uses (or the type of hospital that chooses a specific system) **is** correlated with different quality outcomes. This strongly suggests our `ANOVA` test will be statistically significant.

***Analysis: Research Question 3 (Hospital Type, Adoption, & Outcomes)***

Section `1e` provides two profound answers to our third, most complex question regarding "Critical Access Hospitals" and specialty hospitals.

* **Finding 3a (Adoption): Our hypothesis was wrong.**
    * We hypothesized that smaller, rural `Critical Access Hospitals` would "lag behind" in EHR adoption.
    * The data shows the **opposite**: `Critical Access Hospitals` have an adoption rate of **76.7%**, which is actually *higher* than the `Acute Care Hospitals` (**75.6%**).
    * **Conclusion:** The data refutes our initial assumption. The real story is that certain specialty hospitals (like `Childrens`, `Psychiatric`, and `VA`) show 0% adoption *in this specific dataset*, likely because they do not participate in this particular CMS reporting program.

* **Finding 3b (Outcomes): The "Overall Rating" metric is not universal.**
    * The output for `Childrens` and `Psychiatric` hospitals shows `NaN` for all rating statistics.
    * **Conclusion:** This is a critical data discovery. It explains *why* our `rating_numeric` variable had so many missing values (as seen in `1a`). The "Hospital overall rating" metric **is not applicable** to these specialty hospitals. This is a crucial data limitation to discuss, and it correctly tells us that we cannot use this specific rating to assess their outcomes.

### Inferential Statistics

In [9]:
# 2a. Test for Research Question 1: 'Has_EHR' vs. 'rating_numeric'
# Hypotheses question: Is there a statistically significant difference in mean ratings between hospitals with and without an EHR?
# Null Hypothesis (H0): The mean rating of hospitals with an EHR is the same as those without an EHR. (μ_yes = μ_no)
# Alternative Hypothesis (Hₐ): The mean rating of hospitals with an EHR is different from those without an EHR. (μ_yes ≠ μ_no)
print("--- 2a. Hypothesis Test 1: t-test (Has_EHR vs. Rating) ---")
alpha = 0.05

# Prepare data (remove NaN values)
rating_yes = df[df['Has_EHR'] == 'Yes']['rating_numeric'].dropna()
rating_no = df[df['Has_EHR'] == 'No']['rating_numeric'].dropna()

# Perform t-test
# We set equal_var=False because our descriptive stats (1c) showed different std deviations
t_stat, p_value_ttest = stats.ttest_ind(rating_yes, rating_no, equal_var=False)

print(f"t-statistic: {t_stat:.4f}")
print(f"P-value: {p_value_ttest:.4f}")

if p_value_ttest < alpha:
    print(f"Conclusion: The P-value ({p_value_ttest:.4f}) is LESS than {alpha}. We REJECT the null hypothesis.")
    print(">> There IS a statistically significant difference in mean ratings between the two groups.")
else:
    print(f"Conclusion: The P-value ({p_value_ttest:.4f}) is GREATER than {alpha}. We FAIL TO REJECT the null hypothesis.")
    print(">> There is NO statistically significant difference in mean ratings between the two groups.")

--- 2a. Hypothesis Test 1: t-test (Has_EHR vs. Rating) ---
t-statistic: -1.4463
P-value: 0.1484
Conclusion: The P-value (0.1484) is GREATER than 0.05. We FAIL TO REJECT the null hypothesis.
>> There is NO statistically significant difference in mean ratings between the two groups.


Bad pipe message: %s [b'}\xe4\x9d\x9f\x99\x06z+\x95\xfb%\x93\xf9\xd8x\x9fc\' \x1e\x93N\x01\xe8\xe8\x9eb`\x81a\x19\x0f\x0fQ\xdc\xf4r\xa3\x9d\x98f\x8d\xadP"\xfd7\xd0\xea\xd0t\x00 JJ\x13\x01\x13\x02\x13\x03\xc0+\xc0/\xc0,\xc00\xcc\xa9\xcc\xa8\xc0\x13\xc0\x14\x00\x9c\x00\x9d\x00/\x005\x01\x00\x06s\xaa\xaa\x00\x00\xff\x01\x00\x01\x00\x003\x04\xef\x04\xed\x8a\x8a\x00\x01\x00\x11\xec\x04\xc0;\xd4\xb9\x11\x86a\x9e\xcc\x89\xc7\xe1\xc5\xcb\xd0\x19p\xd2\x18B\x06\xbcy\xac\x8d)\x86\x9bzLT\x0b\x1c#\x02\xc9(\xaf']
Bad pipe message: %s [b'?\x838\xef\xcb\xc2\xcd,a\xd4\xa0K2\xa0\x06 +\x85bS\x1d\nx<*\x88-\r\xb2\x08\xc3\xcb\x1c\xb4\xf9\x02ra\xbe\xd7V\xad\xa1\xebIt\x81\xa0\xcc\xf3Rw\xf6\x08A9\ne\x11\x8c\x8f\xa1\x9bJ\x96\xc8\x90Xz>p\x16"\xc1\x94\xd6\xd4\x87f\xd9\xbd\xb8\x98V\xe8\xc4\xabH7\x02\x94hD]\xda};\xaaX\x02#\x9c\xbf\xe2O\x15@\x1f\xf2J\x9d\xb2Id\x12W\x80\xb3uk\x9a#']
Bad pipe message: %s [b'\xf2W\x06\xccSnN/\xef`\xe8c\x8a\x04\xf1\x99A\xdd [U\xb1!\x82\x12+\x95F\xe5\x00"~\x15\xbd\x80\x9fr9\xd2\xb9\x06\x

***Analysis: t-test (Question 1)***

* **Result:** Our independent samples t-test yielded a **P-value of 0.1484**.

* **Interpretation:** Since our P-value (0.1484) is significantly **greater** than our alpha level (0.05), we **fail to reject the null hypothesis**.

* **Conclusion:** This provides strong statistical evidence that there is **no significant difference** in the mean "Hospital overall rating" between hospitals that have a reported EHR system and those that do not. This result mathematically confirms what we observed visually in our Part 1 boxplot and numerically in our descriptive statistics (section 1c), where the two group means (3.07 vs. 3.14) were nearly identical.

### Comparative analysis 

### Multivariate analysis

### Synthesis

### Reflection